Sources/SafeDICore/Models/Scope.swift

```diff
@@ -78,27 +78,31 @@ final class Scope: Hashable {
 
     func createScopeGenerator(
         for property: Property? = nil,
-        instantiableStack: OrderedSet<Instantiable> = [],
         propertyStack: OrderedSet<Property> = []
     ) throws -> ScopeGenerator {
-        if let cycleIndex = instantiableStack.firstIndex(of: instantiable) {
-            throw ScopeError.dependencyCycleDetected([instantiable] + instantiableStack.elements[0...cycleIndex])
+        if
+            let property,
+            property.propertyType.isConstant,
+            let cycleIndex = propertyStack.firstIndex(of: property)
+        {
+            throw ScopeError.dependencyCycleDetected([property.typeDescription] + propertyStack.elements[0...cycleIndex].map(\.typeDescription))
         } else {
-            var childInstantiableStack = instantiableStack
-            childInstantiableStack.insert(instantiable, at: 0)
             var childPropertyStack = propertyStack
+            let isPropertyCycle: Bool
             if let property {
                 childPropertyStack.insert(property, at: 0)
+                isPropertyCycle = propertyStack.contains(property)
+            } else {
+                isPropertyCycle = false
             }
             let scopeGenerator = ScopeGenerator(
                 instantiable: instantiable,
                 property: property,
-                propertiesToGenerate: try propertiesToGenerate.map {
+                propertiesToGenerate: isPropertyCycle ? [] : try propertiesToGenerate.map {
```

Comment:
we don't need to generate properties if we're fulfilling a cycle – those properties are already fulfilled in the function that fulfills the cycle.